program-runtime/src/loaded_programs.rs

```diff
@@ -819,11 +819,27 @@ impl<FG: ForkGraph> ProgramCache<FG> {
             &entry.program,
             ProgramCacheEntryType::DelayVisibility
         ));
+        fn is_current_env(
+            environments: &ProgramRuntimeEnvironments,
+            env_opt: Option<&ProgramRuntimeEnvironment>,
+        ) -> bool {
```

Comment:
This function always returns `true` during normal operation outside the epoch boundary recompilation phase. That is because all entries which have an environment have the current environment and all others default to `true` directly.

Reply:
It'll be good to add the above as a comment in the code. It's a subtle condition, which will hard to understand a few months from now.